Backend/src/lib/authentication/auth-user.ts

```diff
@@ -29,7 +29,7 @@ export class AuthUser {
   }
 
   async getFullUser(): Promise<User | undefined> {
-    if (this.fullUser || !this.isAuthorized()) {
+    if (this.fullUser ?? !this.isAuthorized()) {
       return this.fullUser;
     }
```

Comment:
```suggestion
    if ( !this.isAuthorized()) {
      return undefined;
    } else if (!!this.fullUser) { // we already cached it before.
      return this.fullUser;
    }
```
first:  The code original I wrote wasn't clear I think. This suggestion is more readable I think.
second: we are doing boolean check here, not assignment. we don't need `??`

Reply:
We don't need to!! Here isn't it, one is enough :
 else if (!!this.fullUser)

Reply:
didn't understand what you mean

Reply:
I mean the line should be like this:
else if (!this.fullUser)
not like that:
else if (!!this.fullUser)